### ISSUE TITLE
espree erroneously complains about arrow function

### ISSUE THREAD

**Issue by tromey** [eslint/espree#145]
This fails to parse, but I think it should succeed:

```
pokyo. node
> require('espree').parse('(a, b=(c)=>c) => {}', {ecmaFeatures: { arrowFunctions: true }})
Error: Line 1: Unexpected token =>
    at throwError (/home/tromey/node_modules/espree/espree.js:2069:17)
    at throwUnexpected (/home/tromey/node_modules/espree/espree.js:2132:5)
    at consumeSemicolon (/home/tromey/node_modules/espree/espree.js:2213:9)
    at parseExpressionStatement (/home/tromey/node_modules/espree/espree.js:3719:5)
    at parseStatement (/home/tromey/node_modules/espree/espree.js:4242:44)
    at parseSourceElement (/home/tromey/node_modules/espree/espree.js:5128:16)
    at parseSourceElements (/home/tromey/node_modules/espree/espree.js:5161:25)
    at parseProgram (/home/tromey/node_modules/espree/espree.js:5181:12)
    at Object.parse (/home/tromey/node_modules/espree/espree.js:5423:19)
    at repl:1:20
```

SpiderMonkey's `Reflect.parse` works on this, though I realize this is not completely definitive.

**Comment by xjamundx** [comment]
Yeah it passes babel's parser as well.

**Comment by xjamundx** [comment]
I'll try to take a stab at this as well...

**Comment by xjamundx** [comment]
```
require('./').parse('(a, b=()=>{}) => {}', {ecmaFeatures: { arrowFunctions: true, defaultParams: true }})
require('./').parse('(a, b=c=>{}) => {}', {ecmaFeatures: { arrowFunctions: true, defaultParams: true }})
require('./').parse('(a, b=c=>c) => {}', {ecmaFeatures: { arrowFunctions: true, defaultParams: true }})
```

These variations do not:

```
 require('./').parse('(a, b=(c,d)=>c) => {}', {ecmaFeatures: { arrowFunctions: true, defaultParams: true }})
require('./').parse('(a, b=(c)=>{}) => {}', {ecmaFeatures: { arrowFunctions: true, defaultParams: true }})
require('./').parse('(a, b=(c)=>c) => {}', {ecmaFeatures: { arrowFunctions: true, defaultParams: true }})
```

**Comment by xjamundx** [comment]
So far no progress, poking around still. Hoping to figure it out.

**Comment by xjamundx** [comment]
Getting closer....the one case is getitng intepretted as a SequenceExpression somewhere..

**Comment by xjamundx** [comment]
Sweet I got a passing test. Super sketchy code, but I'll work on it a bit and get this fixed!
